homeassistant/components/aprilaire/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""The Aprilaire integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant
+
+from .const import DOMAIN, LOG_NAME
+from .coordinator import AprilaireCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry, **kwargs) -> bool:
+    """Set up a config entry for Aprilaire."""
+
+    logger = cast(logging.Logger, kwargs.get("logger"))
+
+    if not logger:  # pragma: no cover
+        logger = logging.getLogger(LOG_NAME)  # pragma: no cover
+
+    config = entry.data
+
+    host = config.get(CONF_HOST)
+    if host is None or len(host) == 0:
+        logger.error("Invalid host %s", host)
+        return False
+
+    port = config.get(CONF_PORT)
+    if port is None or port <= 0:
+        logger.error("Invalid port %s", port)
+        return False
+
+    if coordinator := hass.data.get(DOMAIN, {}).get(entry.unique_id):
```

Comment:
Why would the coordinator already exist? This is the setup, it doesn't

Reply:
@joostlek Yes I think you're correct. I think that originally I had attempted to reuse the coordinator in the config flow, but that's not the case now so I have removed this check.